src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -1867,6 +1986,55 @@ class CreateTableStatementSegment(BaseSegment):
     parse_grammar = match_grammar
 
 
+@tsql_dialect.segment(replace=True)
+class AlterTableStatementSegment(BaseSegment):
+    """An `ALTER TABLE` statement.
+
+    https://docs.microsoft.com/en-us/sql/t-sql/statements/alter-table-transact-sql?view=sql-server-ver15
+    Overriding ANSI to remove TSQL non-keywords MODIFY, FIRST
+    TODO: Flesh out TSQL-specific functionality
```

Comment:
Happy to leave like this for now, looks like it'll require pretty beefy syntax to complete this grammar 👍

Reply:
@jpy-git I submitted #2201 to cover the grammar completion

Reply:
fab thanks 😄